lib/after_commit_action.rb

```diff
@@ -17,6 +17,11 @@ module AfterCommitAction
   end
 
   def execute_after_commit(&block)
+    if ActiveRecord::Base.connection.open_transactions == 0 ||
+      ActiveRecord::Base.connection.add_transaction_record(self).nil?
+      return block.call
+    end
```

Comment:
Yeah you are right. I think I'm going to hit merge on #5 after that test goes green if you're :+1: on that.